packages/paste-core/components/toast/src/Toaster.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import {useTransition, animated} from '@twilio-paste/animation-library';
+import {useTheme} from '@twilio-paste/theme';
+import {Box} from '@twilio-paste/box';
+import {Toast} from './Toast';
+import {ToastPortal} from './ToastPortal';
+import {ToasterProps, ToasterPropTypes} from './types';
+
+export const AnimatedToast = animated(Box);
+
+interface ReturnTargetState {
+  trigger: HTMLElement | null;
+}
+
+const Toaster: React.FC<ToasterProps> = ({toasts, pop, ...props}) => {
+  const [refMap] = React.useState(() => new WeakMap());
+  const [returnTarget, setReturnTarget] = React.useState<ReturnTargetState>({trigger: null});
+  const theme = useTheme();
+
+  const transitions = useTransition(toasts, {
+    from: {height: 0, marginBottom: 0, opacity: 0, transform: 'translateX(100%) scale(1)'},
+    enter: item => async next => {
+      await next({
+        height: refMap.get(item).offsetHeight,
+        marginBottom: theme.space.space40,
```

Comment:
So the marginBottom isn't currently animating correctly because `theme.space.space40` returns a string value with `rem` units, and the initial state is a number (`0`).  To fix this, we should either change this to `40` or the other marginBottoms from `0` to `"0rem"`.  Looks smoother with the change.

Sidebar: we should probably replace the 0 tokens (i.e.: space0) to be a string with the correct unit: `0` -> `"0rem"` to make this type of work tokenizable.